docs/dev/libs/cep.md

```diff
@@ -246,63 +334,399 @@ pattern.where(event => ... /* some condition */).or(event => ... /* or condition
+                @Override
+                public boolean filter(Event value) throws Exception {
+                  return value.getName().equals("b");
+                }
+              });
+              {% endhighlight %}
+              <p>Will generate the following matches for an input sequence: C D A1 A2 A3 D A4 B</p>
+            
+              <p>with consecutive applied: {C A1 B}, {C A1 A2 B}, {C A1 A2 A3 B}</p>
+              <p>without consecutive applied: {C A1 B}, {C A1 A2 B}, {C A1 A2 A3 B}, {C A1 A2 A3 A4 B}</p>
+          </td>
+       </tr>
+       <tr>
+       <td><strong>allowCombinations()</strong><a name="allow_comb_java"></a></td>
+       <td>
+              <p>Works in conjunction with oneOrMore() and times() and imposes non-deterministic relaxed contiguity 
+              between the matching events (as in followedByAny()).</p>
+              <p>If not applied a relaxed contiguity (as in followedBy) is used.</p>
+                   
+              <p>E.g. a pattern like:</p>
+              {% highlight java %}
+              Pattern.<Event>begin("start").where(new SimpleCondition<Event>() {
+                @Override
+                public boolean filter(Event value) throws Exception {
+                  return value.getName().equals("c");
+                }
+              })
+              .followedBy("middle").where(new SimpleCondition<Event>() {
+                @Override
+                public boolean filter(Event value) throws Exception {
+                  return value.getName().equals("a");
+                }
+              }).oneOrMore().allowCombinations()
+              .followedBy("end1").where(new SimpleCondition<Event>() {
+                @Override
+                public boolean filter(Event value) throws Exception {
+                  return value.getName().equals("b");
+                }
+              });
+              {% endhighlight %}
+               <p>Will generate the following matches for an input sequence: C D A1 A2 A3 D A4 B</p>
+               
+               <p>with combinations enabled: {C A1 B}, {C A1 A2 B}, {C A1 A3 B}, {C A1 A4 B}, {C A1 A2 A3 B}, {C A1 A2 A4 B}, {C A1 A3 A4 B}, {C A1 A2 A3 A4 B}</p>
+               <p>without combinations enabled: {C A1 B}, {C A1 A2 B}, {C A1 A2 A3 B}, {C A1 A2 A3 A4 B}</p>
+       </td>
+       </tr>
+  </tbody>
+</table>
 </div>
 
 <div data-lang="scala" markdown="1">
+<table class="table table-bordered">
+    <thead>
+        <tr>
+            <th class="text-left" style="width: 25%">Pattern Operation</th>
+            <th class="text-center">Description</th>
+        </tr>
+    </thead>
+    <tbody>
+      
+        <tr>
+            <td><strong>where(condition)</strong></td>
+            <td>
+              <p>Defines a condition for the current state. Only if an event satisifes the condition, 
+              it can match the state. Multiple consecutive where() clauses lead to their condtions being 
```

Comment:
... can it match the state.

Or turn the sentence around to be more direct:

An event matches the state only if it satisfies the condition.